Source/MaestroVelPred.cpp

```diff
@@ -0,0 +1,1511 @@
+                    }
+                    break;
+                case Outflow:
+                    urx(i,j,k,0) = min(urx(i,j,k,0),0.0);
+                    for (int n = 0; n < AMREX_SPACEDIM; ++n) {
+                        urx(i,j,k,n) = ulx(i,j,k,n);
+                    }
+                    break;
+                case Interior:
+                    break; 
+            }
+
+        // impose hi side bc's
+        } else if (i == ihi+1) {
+            switch (bchi) {
+                case Inflow:
+                    for (int n = 0; n < AMREX_SPACEDIM; ++n) {
+                        ulx(i,j,k,n) = utilde(i,j,k,n);
+                        urx(i,j,k,n) = utilde(i,j,k,n);
+                    }
+                    break;
+                case SlipWall:
+                case Symmetry:
+                    ulx(i,j,k,0) = 0.0;
+                    urx(i,j,k,0) = 0.0;
+                    urx(i,j,k,1) = ulx(i,j,k,1);
+                    break;
+                case NoSlipWall:
+                    for (int n = 0; n < AMREX_SPACEDIM; ++n) {
+                        ulx(i,j,k,n) = 0.0;
+                        urx(i,j,k,n) = 0.0;
+                    }
+                    break;
+                case Outflow:
+                    ulx(i,j,k,0) = max(ulx(i,j,k,0),0.0);
+                    for (int n = 0; n < AMREX_SPACEDIM; ++n) {
+                        urx(i,j,k,n) = ulx(i,j,k,n);
+                    }
+                    break;
+                case Interior:
+                    break;
+            }
+        }
+
+        // No need to compute uimh(:,:,0) since it's equal to utrans-w0
+        // upwind using full velocity to get transverse component of uimhx
+        // Note: utrans already contains w0
+        uimhx(i,j,k,1) = utrans(i,j,k) > 0.0 ? ulx(i,j,k,1) : urx(i,j,k,1);
+        uimhx(i,j,k,1) = fabs(utrans(i,j,k)) < rel_eps ? 
+            0.5*(ulx(i,j,k,1)+urx(i,j,k,1)) : uimhx(i,j,k,1);
+    });
+
+    // y-direction
+    int jlo = domainBox.loVect()[1];
+    int jhi = domainBox.hiVect()[1];
+    bclo = phys_bc[1];
+    bchi = phys_bc[AMREX_SPACEDIM+1];
+
+    AMREX_PARALLEL_FOR_3D(mybx, i, j, k, 
+    {
+        if (ppm_type_local == 0) {
+            Real maxu = max(0.0,ufull(i,j-1,k,1));
+            Real minu = min(0.0,ufull(i,j,k,1));
+            // extrapolate both components of velocity to left face
+            uly(i,j,k,0) = utilde(i,j-1,k,0) + (0.5-(dt2/hy)*maxu)*Imv(i,j-1,k,0);
+            uly(i,j,k,1) = utilde(i,j-1,k,1) + (0.5-(dt2/hy)*maxu)*Imv(i,j-1,k,1);
+            // extrapolate both components of velocity to right face
+            ury(i,j,k,0) = utilde(i,j,k,0) - (0.5+(dt2/hy)*minu)*Imv(i,j,k,0);
+            ury(i,j,k,1) = utilde(i,j,k,1) - (0.5+(dt2/hy)*minu)*Imv(i,j,k,1);
+        } else if (ppm_type_local == 1 || ppm_type_local == 2) {
+            // extrapolate both components of velocity to left face
+            uly(i,j,k,0) = Ipu(i,j-1,k,0);
+            uly(i,j,k,1) = Ipv(i,j-1,k,1);
```

Comment:
see the comments above in the x-direction code -- I think the same applies here.

Reply:
I still think these are wrong.  Since we are butting stuff on the y edges, shouldn't the last index always be `1`?  In the x-direction code it was always `0`

Reply:
yes, I agree with this. I compared to the original Fortran and they should both be 1's